### DISCUSSION TITLE
Scene viewer is not being triggered on Mozilla Firefox

### DISCUSSION THREAD

**Discussion by hd-esteban** [google/model-viewer#2167] [Q&A]
Hi, I'm testing and it seems that Mozilla Firefox 8.1.0 is not executing the Android Intent code to open Scene Viewer.
AR is working properly on chrome with the same device.

**Comment by hd-esteban** [comment]
I made some testing using WebIDE tool and **ARMode** is set as **AR_VIEWER** on Mobile Firefox.
So, when user taps on the AR button **openARViewer** function is executed, but it seems that **undoHashChange** is never executed because Firefox is not able to run the Intent so the web page is not getting redirect to the hash **#model-viewer-no-ar-fallback**

**Comment by cdata** [comment]
Thanks for reporting, I'll take a look!

**Comment by cdata** [comment]
I can confirm that Scene Viewer does not appear to be launching from Firefox on a Pixel 3.

**Comment by hd-esteban** [comment]
I was reading today about it and it seems Intent never worked there.

https://bugzilla.mozilla.org/show_bug.cgi?id=654621
https://bugzilla.mozilla.org/show_bug.cgi?id=1320072

**Comment by cdata** [comment]
@Esteban26 I have recorded video evidence that Scene Viewer can be launched from Firefox, so this definitely worked at one point. I haven't had time yet but I'm going to try a reduced reproduction to see what is going on...

**Comment by Enelar** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1320072 "Wont fix"
Means firefox will never have AR?

**Comment by Enelar** [comment]
#627
